java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/Server.java

```diff
@@ -824,6 +832,24 @@ public void showMessage(MessageParams messageParams) {
             return x;
         }
 
+        @Override
+        public CompletableFuture<String> createTextEditorDecoration(DecorationRenderOptions params) {
+            logWarning(params);
```

Comment:
Shouldn't the methods be abstract? It is strange to see such dummy code in production.

Reply:
I've used the same pattern as there is in `showMessageRequest()`. This class can not be abstract, it's the stub LSP client, which is used only when no LSP client is found (no impl of `NbCodeLanguageClient` is registered).